eo-maven-plugin/src/main/java/org/eolang/maven/OyRemote.java

```diff
@@ -81,4 +86,61 @@ public Input get(final String name) throws MalformedURLException {
         return new InputOf(url);
     }
 
+    /**
+     * Hash of head master.
+     * @param hash String "master"
+     * @return SHA of commit
+     * @checkstyle NonStaticMethodCheck (20 lines)
+     */
+    String getSha(final String hash) {
+        String sha = "master";
+        try {
+            final String query = String.format(
+                // @checkstyle LineLength (1 line)
+                "https://api.github.com/repos/objectionary/home/git/refs/heads/%s",
+                hash
+            );
+            final JSONObject obj = readJsonFromUrl(query);
+            sha = obj.getJSONObject("object").getString("sha");
+        } catch (final IOException | JSONException exception) {
+            Logger.info(this, "Couldn't get commit SHA. It will be set as \"master\"");
+        }
+        return sha;
+    }
+
+    /**
+     * Reader.
+     * @param reader Reader
+     * @return String
+     * @throws IOException if something went wrong
+     */
+    private static String readAll(final Reader reader) throws IOException {
+        final StringBuilder sbuilder = new StringBuilder();
+        int cur;
+        while ((cur = reader.read()) != -1) {
+            sbuilder.append((char) cur);
+        }
+        return sbuilder.toString();
+    }
+
+    /**
+     * Function from reading by url.
+     * @param url URL
+     * @return JSONobject
+     * @throws IOException exception
+     * @throws JSONException exception
+     */
+    private static JSONObject readJsonFromUrl(final String url) throws IOException, JSONException {
+        final InputStream istream = new URL(url).openStream();
+        try {
```

Comment:
@yasamprom I think it would be better to use [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) construction here.